docs/reasons-for-rejection.md

```diff
@@ -1,32 +1,154 @@
+
+#### Third iteration
+
+_Nested JSON_
+
+Then the Flat format was redesigned. The new format of JSON would be stored in the same column `structured_reasons_for_rejection`.
+
+A new type was added:
+
+**V2 `rejection_reasons_type`**
+ - `rejection_reason` - singular
+ - `reasons_for_rejection` - Flat JSON format
+ - `rejection_reasons` - new nested JSON format
+
+Then we stopped using `reasons_for_rejection` (last record in production: 2022-04-26 14:25:40).
+
+[Nested JSON Example](#nested-json)
+
+#### Fourth iteration
+
+_Vendor API Codes_
+
+A new system of providing an API endpoint to vendors which allows them to reject applications based on arbitrary codes we provide to them.
+
+**V3 `rejection_reasons_type`**
+ - `rejection_reason` - singular.
+ - `vendor_api_rejection_reasons` - new nested JSON format assigned by code.
+ - `rejection_reasons` - new nested JSON format.
+
+`vendor_api_rejection_reasons` was introduced recently (first record in production: 2023-06-01 16:32:09).
+
+The codes are stored in a YAML file here [config/rejection_reason_codes.yml](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/rejection_reason_codes.yml).
+
+Some of the changes:
+
+ - https://github.com/DFE-Digital/apply-for-teacher-training/pull/7256
+ - https://github.com/DFE-Digital/apply-for-teacher-training/pull/7257
+ - https://github.com/DFE-Digital/apply-for-teacher-training/pull/7260
+
+
+## Latest
+
 We've iterated the way we capture reasons for rejections several times and this has led to a variety of ways we store the rejection reasons data.
 
 - As a single text field value in `ApplicationChoice#rejection_reason`
 - As a complex set of flat attributes as JSON in `ApplicationChoice#structured_rejection_reasons`
 - As a complex set of nested attributes as JSON in `ApplicationChoice#structured_rejection_reasons`
 
-We use the field and corresponding enum [`ApplicationChoice#rejection_reasons_type`](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/application_choice.rb#L52-L56) to denote the reasons data format.
+We use the field and corresponding enum [`ApplicationChoice#rejection_reasons_type`](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/application_choice.rb#L52-L56) to determine the reasons data format.
+
+|rejection_reasons_type|Column name|data type|Source|Info|
+|---|---|---|---|---|
+|`rejection_reason`|`rejection_reason`|String|Vendor API, ProviderInterface||
+|`reasons_for_rejection`|`structured_rejection_reasons`|flat JSON|ProviderInterface|DEPRECATED 2022|
+|`rejection_reasons`|`structured_rejection_reasons`|nested JSON |ProviderInterface||
+|`vendor_api_rejection_reasons`|`structured_rejection_reasons`|nested JSON|Vendor API|Introduced 2023|
+
+### Why are the old versions still used
+
+The Vendor API is being supported for the lifetime of v1. It is costly for Providers to upgrade if they use an SRS (Student Record System) provided by a 3rd party.
```

Comment:
```suggestion
The Vendor API is being supported for the lifetime of v1. It is costly for Providers to upgrade if they use a SRS (Student Record System) provided by a 3rd party.
```

Reply:
I think an is correct here

Reply:
I've updated the suggested change to put the abbreviation at the end. which is usually how it's written. Feel free to update or leave it, it's very minor.